snuba/web/rpc/common/common.py

```diff
@@ -64,6 +76,32 @@ def transform(exp: Expression) -> Expression:
     query.transform_expressions(transform)
 
 
+def aggregation_to_expression(aggregation: AttributeAggregation) -> Expression:
```

Comment:
@Zylphrex says:

looking this PR, the implementation of all the aggregations on the attribute columns is wrong.
for duration, which is present on every span, it's fine to directly apply the aggregate function over it like avg(duration) but for an attribute, that's wrong because clickhouse uses a default of 0 for spans that dont have that attribute
so avg(attr_num_15[foo]) needs to be to be translated to something like avgIf(attr_num_15[foo], mapContains(attr_num_15, 'foo')) to get the correct aggregation

Reply:
@Zylphrex I created a query processor to do this transformation so as not to pollute the RPC code too much. I also added a test for the case you outlined to make sure it worked as expected

Reply:
Is this something that needs to be toggleable? I would think sometimes you would want to take into account the missing spans as well?

Reply:
can you name an actual case?

Reply:
I would expect this to behave like nulls vs 0s where nulls should be skipped during aggregation and 0s should be included during aggregation.

Reply:
If there's no case to make it toggleable then let's not. Just wanted to know.